src/individual/CalcBaseDamage.c

```diff
@@ -225,836 +310,1421 @@ int CalcBaseDamage(void *bw, struct BattleStruct *sp, int moveno, u32 side_cond,
+        if (defender == sp->rawSpeedNonRNGClientOrder[i]) {
+            // handle Eviolite
+            if (DefendingMon.item_held_effect == HOLD_EFFECT_EVIOLITE) {
+                u16 speciesWithForm;
+                speciesWithForm = PokeOtherFormMonsNoGet(sp->battlemon[defender].species, sp->battlemon[defender].form_no);
+
+                struct Evolution *evoTable;
+                evoTable = sys_AllocMemory(0, MAX_EVOS_PER_POKE * sizeof(struct Evolution));
+                ArchiveDataLoad(evoTable, ARC_EVOLUTIONS, speciesWithForm);
+
+                // If a Pokémon has any evolutions, there should be an entry at the top that isn't EVO_NONE.
+                // In that case, the Pokémon is capable of evolving, and so the effect of Eviolite should apply.
+                if (evoTable[0].method != EVO_NONE) {
+                    defenseModifier = QMul_RoundUp(defenseModifier, UQ412__1_5);
+                }
+
+                sys_FreeMemoryEz(evoTable);
             }
-            if (moveno == MOVE_EARTHQUAKE || moveno == MOVE_MAGNITUDE || moveno == MOVE_BULLDOZE) {
-                damage /= 2;
+
+            // handle Assault Vest
+            if ((DefendingMon.item_held_effect == HOLD_EFFECT_SPDEF_BOOST_NO_STATUS_MOVES) && (movesplit == SPLIT_SPECIAL)) {
+                defenseModifier = QMul_RoundUp(defenseModifier, UQ412__1_5);
             }
-            break;
-        case ELECTRIC_TERRAIN:
-            if (IsClientGrounded(sp, attacker) && movetype == TYPE_ELECTRIC) {
-                damage = damage * 130 / 100;
+
+            // handle Deep Sea Scale
+            if ((DefendingMon.item_held_effect == HOLD_EFFECT_CLAMPERL_SPDEF)
+            && (DefendingMon.species == SPECIES_CLAMPERL)
+            && (movesplit == SPLIT_SPECIAL)) {
+                defenseModifier = QMul_RoundUp(defenseModifier, UQ412__2_0);
             }
-            break;
-        case MISTY_TERRAIN:
-            if (IsClientGrounded(sp, defender) && movetype == TYPE_DRAGON) {
-                damage /= 2;
+
+            // handle metal powder
+            if ((DefendingMon.item_held_effect == HOLD_EFFECT_DITTO_DEF_UP)
+            && (DefendingMon.species == SPECIES_DITTO)
+            && (movesplit == SPLIT_PHYSICAL)) {
+                defenseModifier = QMul_RoundUp(defenseModifier, UQ412__2_0);
+            }
+        }
+    }
+
+    // Apply the chained modifier to the starting defense. That is, multiply the starting defense by the chained defense modifiers, divide by 4096, and pokeRound the result. If the current defense would now be less than 1, make it 1. Finally, if the defense is greater than 65,535, make it the defense modulo 65,536 (defense % 65536). If the defense stat is 0 because of this modifier, the result of base damage will always be 2.
+    calculatedDefense = QMul_RoundDown(calculatedDefense, defenseModifier);
+    calculatedDefense = calculatedDefense < 1 ? 1 : calculatedDefense;
+    calculatedDefense = calculatedDefense % 65536;
```

Comment:
general question for all of the modulo operations--this can still result in a 0 calculatedDefense.  does that mimic future implementations?

Reply:
Yes, there should be logic guarding such cases